extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/RegistrationServiceExtension.java

```diff
@@ -49,12 +69,17 @@ public class RegistrationServiceExtension implements ServiceExtension {
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var monitor = context.getMonitor();
+        var audience = Objects.requireNonNull(context.getSetting(JWT_AUDIENCE_SETTING, null),
+                () -> format("Missing setting %s", JWT_AUDIENCE_SETTING));
+        var authenticationService = new DidJwtAuthenticationFilter(monitor, didPublicKeyResolver, audience);
```

Comment:
@paullatzelsperger @cpeeyush extending the RequestFilter is required in order to inject a header in the request (issuer).

Reply:
@algattik That sounds like the `DidJwtAuthenticationFilter` actually does two things:
 1. verify the signature (= "authenticate")
 2. add a header (i.e. the caller's DID)

In the spirit of "do one thing...": could we indeed convert the filter to an authentication service and use a [`RewriteHandler`](https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/rewrite/handler/RewriteHandler.html) to do the injecting of the DID? We'd have to add some glue code to the `JettyService`.|

As another option we could pass the entire request to the authentication service and have it modify the headers.

Reply:
@paullatzelsperger One more possible idea is that EDC-AuthenticationService may return a List of AuthenticatedPrincipal or Principal and then EDC-AuthenticationRequestFilter can use them to add further information to into headers.

But this will require change in EDC code, so will that be okay if we do this separately in the next PR and merge this one as is for now ? If we agree then I will open a separate github issue on EDC repo to cover this in an upcoming PR.

Reply:
That's what I mean by leaky abstraction. I don't see much point in creating an overlay for the ContainerRequestFilter that has already been carefully designed in JAX-RS as a standard, especially if we are going to use JAX-RS objects in the interface...

Reply:
The advantage of having _one_ requst filter, and _several_ `AuthenticationService` instances is to make use of the DI mechanism. Implementors of a particular authentication method need only have an `@Provider` method in their extension.

But for the time being I'm good with leaving the `DidAuthenticationFilter` as-is and merge it.